File: backend/api-server/src/main/java/com/neogulmap/neogul_map/repository/UserRepository.java
package com.neogulmap.neogul_map.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import com.neogulmap.neogul_map.domain.User;

public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByEmail(String email);
    
    // 프로필 이미지가 있는 사용자 조회
    Optional<User> findByProfileImageIsNotNull();
    
    // 특정 OAuth 제공자의 사용자 조회
    Optional<User> findByOauthProviderAndOauthId(String oauthProvider, String oauthId);
}

File: app/src/main/java/com/example/neogulmap/domain/repository/UserRepository.kt
package com.example.neogulmap.domain.repository

import com.example.neogulmap.data.api.UserRequest
import java.io.File

interface UserRepository {
    /**
     * Creates a user on the backend, typically after a successful social login.
     *
     * @param userRequest The user data obtained from the social login provider.
     * @param profileImage An optional profile image file.
     * @return A Result wrapper indicating success or failure.
     */
    suspend fun createUser(userRequest: UserRequest, profileImage: File?): Result<Unit>
}
